components/com_content/models/article.php

```diff
@@ -164,7 +177,7 @@ public function getItem($pk = null)
 				}
 
 				// Check for published state if filter set.
-				if (((is_numeric($published)) || (is_numeric($archived))) && (($data->state != $published) && ($data->state != $archived)))
+				if ( !($haveToken == $currentToken) && ((is_numeric($published)) || (is_numeric($archived))) && (($data->state != $published) && ($data->state != $archived)))
```

Comment:
this means that token check would be performed only if status=0 else no need for token check[as article is already published].....and then rest of the code?